pyomo/gdp/plugins/chull.py

```diff
@@ -210,231 +233,205 @@ def _apply_to_impl(self, instance, **kwds):
-        # the containers that don't have any active guys inside of
-        # them. So the invalid component logic will tell us if we
-        # missed something getting transformed.
-        for obj in transBlock.disjContainers:
-            if not obj.active:
-                continue
-            for i in obj:
-                if obj[i].active:
-                    break
-            else:
-                # HACK due to active flag implementation.
-                #
-                # Ideally we would not have to do any of this (an
-                # ActiveIndexedComponent would get its active status by
-                # querring the active status of all the contained Data
-                # objects).  As a fallback, we would like to call:
-                #
-                #    obj._deactivate_without_fixing_indicator()
-                #
-                # However, the sreaightforward implementation of that
-                # method would have unintended side effects (fixing the
-                # contained _DisjunctData's indicator_vars!) due to our
-                # class hierarchy.  Instead, we will directly call the
-                # relevant base class (safe-ish since we are verifying
-                # that all the contained _DisjunctionData are
-                # deactivated directly above).
-                ActiveComponent.deactivate(obj)
-
         # HACK for backwards compatibility with the older GDP transformations
         #
-        # Until the writers are updated to find variables on things
-        # other than active blocks, we need to reclassify the Disjuncts
-        # as Blocks after transformation so that the writer will pick up
-        # all the variables that it needs (in this case, indicator_vars).
+        # Until the writers are updated to find variables on things other than
+        # active blocks, we need to reclassify the Disjuncts as Blocks after
+        # transformation so that the writer will pick up all the variables that
+        # it needs (in this case, indicator_vars and also variables which are
+        # declared in a single Disjunct and only used on that Disjunct (as they
+        # will not be disaggregated)).
         if _HACK_transform_whole_instance:
             HACK_GDP_Disjunct_Reclassifier().apply_to(instance)
 
+    def _add_transformation_block(self, instance):
+        # make a transformation block on instance where we will store
+        # transformed components
+        transBlockName = unique_component_name(
+            instance,
+            '_pyomo_gdp_chull_relaxation')
+        transBlock = Block()
+        instance.add_component(transBlockName, transBlock)
+        transBlock.relaxedDisjuncts = Block(Any)
+        transBlock.lbub = Set(initialize = ['lb','ub','eq'])
+        # We will store all of the disaggregation constraints for any
+        # Disjunctions we transform onto this block here.
+        transBlock.disaggregationConstraints = Constraint(Any)
```

Comment:
Now that we can, I think these should be declared as `Constraint(NonNegativeIntegers, Any)`

Reply:
Ummm, for reasons I no longer remember, we index these with the original constraint index followed by an integer... So I guess this could be `Constraint(Any, NonNegativeIntegers)`, but does that provide any validation actually? (Or we could change the order and make what you said true.)

Reply:
With the set rewrite, both will work and provide the same validation (one that the index starts with a nonnegative integer, and the other that it ends with one).  I tend to like to have the non-dimensioned Set be at the end, though (things print nicer).